nautobot/extras/models/datasources.py

```diff
@@ -24,7 +24,11 @@ class GitRepository(PrimaryModel):
         max_length=100,
         unique=True,
     )
-    slug = AutoSlugField(populate_from="name")
+    slug = AutoSlugField(
+        populate_from="name",
+        help_text="Internal field name. Please use underscores rather than dashes in this key.",
+        slugify_function=slugify_dashes_to_underscores,
+    )
 
     remote_url = models.URLField(
         max_length=255,
```

Comment:
Comparing against #3426, isn't this missing `clean()` logic to enforce validation of the slug?

Reply:
My bad, I assumed the field did the validation. Adding.